### ISSUE TITLE
leksah itself in sandbox

### ISSUE THREAD

**Issue by flip111** [leksah/leksah#189]
Is it possible to install leksah itself in a sandbox? so that the cabal dependencies are not global.

Also is it possible to build from source, maybe this can be added to the readme?

**Comment by hamishmack** [comment]
> Is it possible to install leksah itself in a sandbox? so that the cabal dependencies are not global.

In short yes, but I don't think we have step by step instructions.  I think it is just a matter of running `cabal sandbox init` and adding `.cabal-sandbox/bin` to your `PATH` before running any `cabal install`.  

> Also is it possible to build from source, maybe this can be added to the readme?

https://github.com/leksah/leksah#building-from-source

**Comment by flip111** [comment]
Sorry about question two, i skipped right to https://github.com/leksah/leksah#building-on-linux assuming it was not building anything there because there is no build command.

For the sandbox typing `cabal sandbox init` does not create a `bin` directory in `.cabal-sandbox`

**Comment by hamishmack** [comment]
> For the sandbox typing cabal sandbox init does not create a bin directory in .cabal-sandbox

It should be created when you install a package with an executable.

**Comment by flip111** [comment]
This message about packages getting broken (at the bottom) makes me believe the sandbox is not really working as it should

```
~/leksah$ cabal install
Resolving dependencies...
In order, the following would be installed:
ansi-terminal-0.6.2.2 (new package)
base-orphans-0.4.4 (new package)
bytestring-builder-0.10.6.0.0 -bytestring_has_builder (new package)
dlist-0.7.1.2 (new package)
executable-path-0.0.3 (new package)
extra-1.4.1 (new package)
ghc-paths-0.1.0.9 (new package)
ghcjs-codemirror-0.0.0.1 (new package)
haddock-2.13.2.1 (latest: 2.16.1) (new package)
haskell-lexer-1.0 (new package)
hscolour-1.23 (new package)
hxt-charproperties-9.2.0.1 (new package)
hxt-unicode-9.0.2.4 (new package)
network-2.6.2.1 (new version)
prelude-extras-0.4 (new package)
pretty-show-1.6.8.2 (new package)
stm-2.4.4 (new version)
strict-0.3.2 (new package)
tagged-0.8.1 (new package)
reflection-2 (new package)
text-1.2.1.3 -integer-simple (new version)
blaze-builder-0.4.0.1 (new package)
blaze-markup-0.7.0.2 (new package)
blaze-html-0.8.1.0 (new package)
hashable-1.2.3.3 (new version)
nats-1 (new package)
polyparse-1.11 (new package)
cpphs-1.19.3 (new package)
haskell-src-exts-1.16.0.1 (new package)
scientific-0.3.3.8 (new package)
transformers-0.4.3.0 (new version)
QuickCheck-2.8.1 -base4point8 (new version)
StateVar-1.1.0.1 (new package)
attoparsec-0.13.0.1 (new version)
cmdargs-0.10.13 (new package)
mmorph-1.0.4 (new package)
mtl-2.2.1 (new version)
binary-shared-0.8.3 (new package)
hslogger-1.2.9 (new package)
parsec-3.1.9 (new version)
hxt-regex-xmlschema-9.2.0.2 (new package)
network-uri-2.6.0.3 (new package)
HTTP-4000.2.20 (new version)
hxt-9.3.1.15 +network-uri (new package)
regex-base-0.93.2 (reinstall) changes: mtl-2.1.2 -> 2.2.1
regex-tdfa-1.2.1 (new package)
regex-tdfa-text-1.0.0.3 (new package)
streaming-commons-0.1.12.1 +use-bytestring-builder (new package)
transformers-compat-0.4.0.4 (new package)
distributive-0.4.4 (new package)
exceptions-0.8.0.2 (new package)
transformers-base-0.4.4 (new package)
monad-control-1.0.0.4 (new package)
lifted-base-0.2.3.6 (new package)
resourcet-1.1.6 (new package)
unordered-containers-0.2.5.1 (new version)
aeson-0.7.0.6 (latest: 0.9.0.1) (new package)
semigroups-0.16.2.2 (new package)
bifunctors-5 (new package)
shakespeare-2.0.5 (new package)
uniplate-1.6.12 (new package)
hlint-1.9.21 (new package)
utf8-string-1.0.1.1 (new package)
cairo-0.13.1.0 (new package)
glib-0.13.2.1 (new package)
gio-0.13.1.0 (new package)
pango-0.13.1.0 (new package)
gtk3-0.14.1 (new package)
gtksourceview3-0.13.2.0 (new package)
ltk-0.15.0.4 (new package)
vado-0.0.5 (new package)
vcswrapper-0.1.1 (latest: 0.1.2) (new package)
vcsgui-0.1.3.0 (new package)
void-0.7 (new package)
conduit-1.2.5 (new package)
attoparsec-conduit-1.1.0 (new package)
conduit-extra-1.1.9.1 (new package)
contravariant-1.3.2 (new package)
comonad-4.2.7.2 (new package)
leksah-server-0.15.0.9 (new package)
profunctors-5.1.1 (new package)
semigroupoids-5.0.0.4 (new package)
free-4.12.1 (new package)
adjunctions-4.2.1 (new package)
kan-extensions-4.2.2 (new package)
lens-4.12.3 (new package)
webkitgtk3-0.14.1.0 (new package)
ghcjs-dom-0.2.1.0 (new package)
webkitgtk3-javascriptcore-0.13.1.0 (new package)
jsaddle-0.2.1.0 (new package)
leksah-0.15.1.5 (new package)
cabal: The following packages are likely to be broken by the reinstalls:
regex-posix-0.95.2
regex-compat-0.95.1
Use --force-reinstalls if you want to install anyway.
```

**Comment by jaccokrijnen** [comment]
I just tried this:

```
git clone https://github.com/leksah/leksah.git
cd leksah
cabal sandbox init
cabal update
cabal install Cabal
cabal install alex happy
cabal install gtk2hs-buildtools
cabal install

export PATH=~/leksah/.cabal-sandbox/bin:$PATH    # you should put this in your ~/.profile file:
.cabal-sandbox/bin/leksah
```

Could you try and see if doing this works? Then I will add it to the readme

**Comment by flip111** [comment]
@jaccokrijnen at `cabal install` i still get the same message as

**Comment by jaccokrijnen** [comment]
What are your
```
cabal --version
```
and
```
ghc --version
```
?

**Comment by flip111** [comment]
cabal-install version 1.20.0.3
using version 1.20.0.2 of the Cabal library

The Glorious Glasgow Haskell Compilation System, version 7.6.3

**Comment by jaccokrijnen** [comment]
Leksah is only available for ghc 7.8.3 and higher, so it would be best to upgrade. If you are on Ubuntu, https://launchpad.net/~hvr/+archive/ubuntu/ghc is a nice place for getting up-to-date ghc and cabal. Let me know if this helps!

**Comment by flip111** [comment]
I just installed haskell platform, i will try again at the new release, thanks!
